translator/GCP_cloud_function/cloud_function/main.py

```diff
@@ -1,35 +1,69 @@
-# {'bucket': 'icone_files', 'contentType': 'image/jpeg', 'crc32c': 'ZO4N6A==', 'etag': 'CNu+up7tnu4CEAE=', 
-# 'generation': '1610745138552667', 'id': 'icone_files/61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG/1610745138552667', 
-# 'kind': 'storage#object', 'md5Hash': 'yOlwCtFi0L/3GPe23rYGbw==', 
-# 'mediaLink': 'https://www.googleapis.com/download/storage/v1/b/icone_files/o/61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG?generation=1610745138552667&alt=media', 
-# 'metageneration': '1', 'name': '61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG',
-#  'selfLink': 'https://www.googleapis.com/storage/v1/b/icone_files/o/61862931_1039122999618882_7010636350885986304_o_1039122996285549.JPEG',
-#   'size': '447421', 'storageClass': 'STANDARD', 'timeCreated': '2021-01-15T21:12:18.552Z', 'timeStorageClassUpdated': '2021-01-15T21:12:18.552Z', 
-#   'updated': '2021-01-15T21:12:18.552Z'}
-
 import icone_translator
 from gcloud import storage
 import json
 import xmltodict
+import shutil
+import urllib.request as request
+from contextlib import closing
+from google.cloud import pubsub_v1
+import os
+from google.cloud import secretmanager
+
+
+
+server = os.environ['ftp_server_address']
+port = os.environ['ftp_port']
+user=os.environ['icone_ftp_username']
+password=os.environ['icone_ftp_password']
+#user,password=get_ftp_credentials()
+filepath = os.environ['ftp_icone_file_path']
+
+
+ftpString = 'ftp://{0}:{1}@{2}:{3}/{4}'
+ftpUrl = ftpString.format(user, password, server, port, filepath)
+
+
+def translate_newest_icone_to_wzdx(event, context):
+
+
+  icone_data=get_ftp_file(ftpUrl)
+  icone_obj=parse_xml(icone_data)
+  info=icone_translator.initialize_info()
+
+  wzdx_obj=icone_translator.wzdx_creator(icone_obj,info)
+  wzdx_schema=get_wzdx_schema('wzdx_schema.json')
+  #this will throw an exception if validation fails
+  icone_translator.validate_wzdx(wzdx_obj,wzdx_schema)
+  print(json.dumps(wzdx_obj))
+
+  publisher = pubsub_v1.PublisherClient()
+  topic_path = publisher.topic_path(os.environ['project_id'], os.environ['wzdx_topic_id'])
+  future=publisher.publish(topic_path,str.encode(json.dumps(wzdx_obj, indent=2)),origin='auto_icone_translator_ftp cloud function')
+  print(future.result())
+  return
+
+def get_ftp_file(url) :
+  with closing(request.urlopen(url)) as r:
+    return r.read().decode('utf-8-sig')
+
+
+def parse_xml(xml_string):
+  return xmltodict.parse(xml_string)
+
+def get_wzdx_schema(schema_file_name):
+  return json.loads(open(schema_file_name,'r').read())
+
+def get_ftp_credentials():
```

Comment:
Is this function needed? It looks like the call to it is commented out above and you're accessing the properties directly instead.

Reply:
we pre-created it for when we have secrets working and we are going to start using this function when we get permissions to use secrets.

Reply:
Let's at least add a comment to the start of the function explaining that it is a function to parse out secrets, and is to be implemented.

Reply:
I commented it in code.Thanks